schedstat.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2019 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package procfs
+
+import (
+	"bufio"
+	"os"
+	"regexp"
+	"strconv"
+)
+
+var (
+	cpuLineRE = regexp.MustCompile(`cpu(\d+) (\d+) (\d+) (\d+) (\d+) (\d+) (\d+) (\d+) (\d+) (\d+)`)
+)
+
+// Schedstat contains scheduler statistics from /proc/schedstats
+//
+// See
+// https://github.com/torvalds/linux/blob/master/Documentation/scheduler/sched-stats.txt
+// for a detailed description of what these numbers mean.
+type Schedstat struct {
+	CPUs []*SchedstatCPU
+}
+
+// SchedstatCPU contains the values from one "cpu<N>" line
+type SchedstatCPU struct {
+	CPUNum string
+
+	RunningJiffies uint64
+	RunningSeconds float64
```

Comment:
What do you think about adding a method to dynamically provide the seconds instead of storing both jiffies and seconds in the struct?

```golang
func (s SchedstatCPU) RunningSeconds() float64 {
	return float64(s.RunningJiffies) / userHZ
}
```